server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/model/MockSMTPBehaviorInformationTest.java

```diff
@@ -0,0 +1,169 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mock.smtp.server.model;
+
+import static org.apache.james.mock.smtp.server.model.MockSMTPBehaviorInformation.RemainingAnswersCounter;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.apache.james.mock.smtp.server.model.MockSMTPBehaviorInformation.AnswersCounter;
+import org.apache.james.mock.smtp.server.model.MockSMTPBehaviorInformation.UnlimitedAnswersCounter;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+
+class MockSMTPBehaviorInformationTest {
+
+    @Test
+    void shouldMatchBeanContract() {
+        EqualsVerifier.forClass(MockSMTPBehaviorInformation.class)
+            .suppress(Warning.NULL_FIELDS)
+            .withIgnoredFields("remainingAnswersCounter")
+            .withPrefabValues(RemainingAnswersCounter.class, new UnlimitedAnswersCounter(), new UnlimitedAnswersCounter())
```

Comment:
do this because `MockSMTPBehaviorInformation.equals()` use delegated methods from an Abstract type. It's well noted in EqualsVerifier documentation. I have to use `withPrefabValues` to inject some concrete implementation of the Abstract type.